ssl/statem/extensions_clnt.c

```diff
@@ -1212,6 +1215,18 @@ EXT_RETURN tls_construct_ctos_psk(SSL_CONNECTION *s, WPACKET *pkt,
             goto dopsksess;
         }
 
+#ifndef OPENSSL_NO_ECH
+        /*
+         * When doing ECH, we get here twice (for inner then outer). The
+         * 2nd time (for outer) we can skip some checks as we know how
+         * those went last time.
+         */
+        if (s->ext.ech.es != NULL && s->ext.ech.ch_depth == 0) {
+            dores = (s->ext.tick_identity > 0);
```

Comment:
`s->ext.tick_identity` also gets incremented in the case that `s->psksession != NULL`. So I think this condition is a bit more complicated. Something like this:
```` c
dores = (s->psksession == NULL) ? s->ext.tick_identity > 0 : s->ext.tick_identity > 1;
````

Reply:
Oh...and we reset `tick_identity` at the beginning of this function. That's going to need some adjustment.

Reply:
urgh, ok gonna test this one before trying again:-) more in a bit...

Reply:
Ok, I've added the `tick_identity` to the ECH state, made a test for that and this seems to work ok. See what you think of this one.

The test script and trace output are in a [gist](https://gist.github.com/sftcd/f372483606215230c72fabbd29869bf6)